netty/src/main/java/io/grpc/transport/netty/NettyClientHandler.java

```diff
@@ -448,7 +459,11 @@ private NettyClientStream clientStream(Http2Stream stream) {
     return stream.getProperty(streamKey);
   }
 
-  private int getAndIncrementNextStreamId() {
+  private int getAndIncrementNextStreamId() throws StatusException {
+    if (nextStreamId < 0) {
+      throw EXHAUSTED_STREAMS_STATUS.asException();
```

Comment:
Don't we also need to close the channel? Otherwise we will continually fail future RPCs. It seems that might should happen if `nextStreamId < 0` at the _end_ of this method.